src/main/java/org/jabref/gui/externalfiles/FileFilterUtils.java

```diff
@@ -0,0 +1,111 @@
+package org.jabref.gui.externalfiles;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileFilterUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileFilterUtils.class);
+    
+    /* Returns the last edited time of a file as LocalDateTime. */
+    public static LocalDateTime getFileTime(Path path) {
+        FileTime lastEditedTime = null;
+        try {
+            lastEditedTime = Files.getLastModifiedTime(path);
+        } catch (IOException e) {
+            LOGGER.error("Could not retrieve file time", e);
+        }
+        LocalDateTime localDateTime = lastEditedTime
+                .toInstant()
```

Comment:
IntelliJ complains here, that toInstant may throw a NPE. This seems possible, since you don't return when a IOException occurs. Is that something we should think about?

Reply:
Thank you very much @calixtus , so maybe we need to place the code in a try catch-block to handle the NPE?

Reply:
Well, there must be a value returned from this method... Which one, if getLastModifiedTime fails?
Today? null?

Reply:
Yes, sorry you are right, we will return LocalDateTime.now() when an IOException occurs